### PR TITLE
Addressing Issue199 dictionary handling

### PR DESCRIPTION
**Pull request by bannanc**
@limn1 ran the smirky example in python3 today and found a bug in the score_utils functions, see issue #199 . We didn't have any tests for those. They used to use dict.has_key() which has been removed. I replaced those lines with the `in` usage and created a test for the three methods in the script.

### CODE REVIEW

**Comment by bannanc** [comment]
@davidlmobley they're pretty small changes, do you have time for a quick code review, I can ask someone else if you don't.

**Comment by davidlmobley** [comment]
Reviewing, yes. Note, @bannanc , that it is failing tests.

**Review by davidlmobley — APPROVED** [review]
Looks good to me once tests pass, @bannanc . (I didn't check if the failure was something introduced by this change, though.)

One note: Did you intend to add the `test_smirky.pdf` file to the repo? Naively it seems unnecessary/distracting.

**Comment by bannanc** [comment]
I'll check what tests it's failing in a little bit and update gitignore to
not store the pdf.

**Comment by bannanc** [comment]
@davidlmobley it looks like there error is because it can't find the "test_smirky.csv" trajectory file I put in the test folder. Should I instead put it somewhere in the `data` directory instead?

**Comment by davidlmobley** [comment]
Yes, @bannanc - if you have a file you need to process to conduct a test, it should go in `data` somewhere.

**Comment by bannanc** [comment]
@davidlmobley 
Here is the current error:
```
======================================================================
ERROR: Failure: ImportError (No module named pylab)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/nose/loader.py", line 407, in loadTestsFromName
    module = resolve_name(addr.module)
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/nose/util.py", line 312, in resolve_name
    module = __import__('.'.join(parts_copy))
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/smarty/__init__.py", line 17, in <module>
    from smarty.score_utils import *
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/smarty/score_utils.py", line 3, in <module>
    import pylab as pl
ImportError: No module named pylab
```

the smarty_sampler does something similar so this makes me think the tests for SMARTY don't include the plotfile option.

**Comment by davidlmobley** [comment]
That's probably the case, @bannanc . Just need to update the installation stuff (as per Slack) to require matplotlib. Thanks.

**Comment by bannanc** [comment]
I'm also working on updating the smarty_sampler tests. I'm switching this back to WIP until I can get this fixed.

File: devtools/conda-recipe/meta.yaml [hunk 1]
```diff
@@ -21,6 +21,8 @@ requirements:
     - lxml
     - openmoltools >=0.7.3
     - parmed
+    - pylab
```
Comment (davidlmobley) [inline devtools/conda-recipe/meta.yaml]: @bannanc - you want `matplotlib`; there's no `pylab` package.

File: devtools/travis-ci/install.sh [hunk 2]
```diff
@@ -20,6 +20,8 @@ conda update --yes conda
 conda install --yes conda-build jinja2 anaconda-client pip
 conda install --yes -c omnia openmoltools
 conda install --yes -c omnia parmed
+conda install --yes -c matplotlib
+conda install --yes pandas=0.13.1
```
Comment (davidlmobley) [inline devtools/travis-ci/install.sh]: @bannanc - maybe the attempt to ask for pandas=0.13.1 is causing problems? I'm getting an "unsatisfiable dependencies" issue on my laptop if I do it this way, whereas just asking for pandas works fine. (I get 0.19.2).

**Comment by bannanc** [comment]
I tried a number of variations with pandas and I couldn't get it to work, I
started with the basic "conda install --yes pandas". The warning on github
still says no module pandas exists no matter what. I'll fix the
pylab/matplotlib tomorrow.

File: devtools/conda-recipe/meta.yaml (hunk 1 above)
Comment (bannanc) [reply devtools/conda-recipe/meta.yaml]: When I have matplotlib there I get this error in the Travis test:
```
Test FragmentSampler runs for 10 iterations with no failures ... QXcbConnection: Could not connect to display 
/home/travis/build.sh: line 59:  1702 Aborted                 (core dumped) nosetests -vv --nocapture --with-timer $PACKAGENAME
```
I get pylab is a module and not the package, but that is what is working...

**Comment by bannanc** [comment]
During the travis test it says:
```
The following NEW packages will be INSTALLED:
    mkl:             2017.0.1-0        
    numpy:           1.11.3-py27_0     
    openssl:         1.0.2k-0          
    pandas:          0.19.2-np111py27_1
...
```
but then later says there is no module pandas, I'm at a loss on what else I need to do.

**Comment by bannanc** [comment]
Many of the other github repositories I'm looking at use the following line in their install.sh
```
    conda create -n testenv --yes python=$PYTHON_VERSION numpy=$NUMPY_VERSION scipy=$SCIPY_VERSION pandas libgfortran
```

**Comment by bannanc** [comment]
@davidlmobley 
Even if we get pandas working correctly there are some issues with the new test_sampler.py script that I'll need to fix. I discovered them while testing my changes on the other pull request.

**Comment by bannanc** [comment]
I'm getting an error about connection to display that I'm not familiar with, it seems to happen in the section of the script that makes the plot output.
@davidlmobley do you know what would cause this?
```
QXcbConnection: Could not connect to display 
/home/travis/build.sh: line 59:  1702 Aborted                 (core dumped) nosetests -vv --nocapture --with-timer $PACKAGENAME
```

**Comment by davidlmobley** [comment]
@bannanc - responded in Slack, sorry. I think this is caused by `matplotlib`/`pylab` trying to connect to the display, and you can deal with it by selecting a "backend" which will only output to files and not connect to display, such as `pdf`.

**Comment by davidlmobley** [comment]
@bannanc - looks like that fixed that problem and now you only have issues with initializing for tests, I think.

**Comment by bannanc** [comment]
@davidlmobley Finally everything is passing, there are now more script changes to the test_sampler.py script, would you mind doing another code check?

**Review by davidlmobley — APPROVED** [review]
I don't see any issues, @bannanc . Are you ready to merge?

**Comment by bannanc** [comment]
yes, merging now!

**Pull request merged**